### ISSUE TITLE
Is someone updating this repo but in another github repo/profile?

### ISSUE THREAD

**Issue by terremoth** [andlabs/libui#547]
This seems almost dead and @andlabs doesn't want patreon support (#486) or he never comes to see its repos here.

**Comment by rubyFeedback** [comment]
I think the bigger issue is in regards to long-term maintenance and improvements. Ideally we could find to see to
distribute the work load onto several maintainers who keep the spirit of libui alive (cross-platform focus first, 
although I'd also love to see a SLIGHTLY less stringent reservation in regards to additional, custom modifications;
for instance I'd love to use CSS at the least when using libui on linux, which should be possible via GTK +
CssProvider - no idea if that is possible on windows, but it would be nice as an option).

Looking at the peak activity of libui, I think andlabs was most active from 2015 to 2017 or so. He did add more
code and code changes past that point, but I think one issue is that it is a lot a one-person project, which is not
ideal. Sadly there is no way for me to maintain libui as such - I simply don't know C and C++ well enough.

Perhaps there could be some additional repository aka "patches to libui" to add individual functionality,
and then somehow organize these patches to make regular releases, even if andlabs does not have the
time or motivation (which is also ok, by the way, I completely understand that time constraints exist,
lack of motivation, funding and so forth, all ok too).

If there is indeed some github repo for additional libui patches, this would be nice to know and see. htop 
maintained by Hisham eventually had new maintainers after a set of patches was not added for a longer
time, so perhaps this model could also work for libui. I guess it needs some initial people to drive forward
though, and these need to know C or C++ fairly well I think. And probably some cross-platform way, 
although other users could help test things, if a call for such testing is made I am sure.

**Comment by terremoth** [comment]
> "distribute the work load onto several maintainers"

The problem is: can be more contributors but the repo maintainer is only one. 
Only if what I told was done: someone forked and continues, but I think that didn't happen (at least not yet). 
Let's remember also, GTK is on his 4th version now. 

> even if andlabs does not have the
time or motivation (which is also ok, by the way, I completely understand that time constraints exist,
lack of motivation, funding and so forth, all ok too).

indeed, no problem at all (seriously) with someone does not caring anymore or if cannot work anymore on a project, the problem is indeed if it dies and nobody continues and github has no option in this url: https://github.com/andlabs/libui/network/members to tell if other forks are more updated... that could be a useful feature they could implement 

One of my dreams is to see a working good libui working on PHP so everyone could build GUI desktop apps with it. 
The other projects: PHP-GTK, PHP-TK, wxPHP and PHP-QT are all dead or forgotten.

**Comment by rubycoder** [comment]
>>  github has no option in this url:
https://github.com/andlabs/libui/network/members to tell if other forks are
more updated

I think this url:  https://github.com/andlabs/libui/network
shows that there are 5 (or so) other repos that have newer commits than the
andlabs one

Wayne

**Comment by terremoth** [comment]
Indeed, thank you! I found a repo that has a 22 minutes ago commit that is updating!!

https://github.com/DonMartio/libui

**Comment by DonMartio** [comment]
Hi there,
just to clarify. Although my corporation is pretty interested in this project, and the corresponding php-ui bindings, i am just scratching on the surface at the moment. So don't expect to much right now. Furthermore i have just linux here (and my colleges too) so we don't have the resources to test for mac or windows. 
However, it would be really sad if this project would cease to exist.

**Comment by rubycoder** [comment]
>> Indeed, thank you!
You're welcome!

Also, scrolling back in time on https://github.com/andlabs/libui/network
shows a number of other repos that have commits that just happen to have
been made before the last andlabs commit (1/2/2021)

**Comment by terremoth** [comment]
@rubycoder probably are PULL REQUESTS not merged

**Comment by cody271** [comment]
I propose that it is now time we try to move libui forward with a completely new fork.

Please see the new org with repo here: [libui-ng/libui-ng](https://github.com/libui-ng/libui-ng)

**Comment by terremoth** [comment]
> I propose that it is now time we try to move libui forward with a completely new fork.
> 
> Please see the new org with repo here: [libui-ng/libui-ng](https://github.com/libui-ng/libui-ng)

So you're going to continue it?

**Comment by DonMartio** [comment]
May we all are.

**Comment by terremoth** [comment]
> May we all are.

I can test things, but I am not a C/C++ programmer. 
I can also fund/help with a some money, if someone starts seriously work on it.

**Comment by DonMartio** [comment]
I have no problems with C/C++ in this matter but i have no insights in mac or windows world.

**Comment by terremoth** [comment]
> I have no problems with C/C++ in this matter but i have no insights in mac or windows world.

That's awesome already, because if you can at least fix the Linux part, other people from other OS's will come to help, or in THE WORST SCENARIO: you will make possible making UI apps again on linux with this repo +updated!

**Comment by cody271** [comment]
> > I propose that it is now time we try to move libui forward with a completely new fork.
> > Please see the new org with repo here: [libui-ng/libui-ng](https://github.com/libui-ng/libui-ng)
> 
> So you're going to continue it?

As I'm the [contributor](https://github.com/andlabs/libui/graphs/contributors) to libui listed after @andlabs, the responsibility to continue appears to fall on me now yes.

**Comment by terremoth** [comment]
> > > I propose that it is now time we try to move libui forward with a completely new fork.
> > > Please see the new org with repo here: [libui-ng/libui-ng](https://github.com/libui-ng/libui-ng)
> > 
> > 
> > So you're going to continue it?
> 
> As I'm the [contributor](https://github.com/andlabs/libui/graphs/contributors) to libui listed after @andlabs, the responsibility to continue appears to fall on me now yes.

Good. Maybe you can start fixing some issues and analyze some pull requests...

**Comment by cody271** [comment]
>
> Good. Maybe you can start fixing some issues and analyze some pull requests...

Yes please file all issues you'd like fixed now to the new repo going forward. Ideally with a reference to the old issue, if applicable. PRs welcome!

**Comment by terremoth** [comment]
> > Good. Maybe you can start fixing some issues and analyze some pull requests...
> 
> Yes please file all issues you'd like fixed now to the new repo going forward. Ideally with a reference to the old issue, if applicable. PRs welcome!

I think it is a nice start here: https://github.com/andlabs/libui/milestones
@andlabs created the "Alpha 5" Milestone that he discussed many issues, so there could be a good start because there are more conversation related and reported. What do you think?

**Comment by terremoth** [comment]
Maybe @DonMartio can also help you there

**Comment by cody271** [comment]
>
> What do you think?

@terremoth I will be prioritizing all issues by actual present users need, so please manually reopen any issues you personally care about from that list in the new repo. Thanks.

**Comment by terremoth** [comment]
> > What do you think?
> 
> @terremoth I will be prioritizing all issues by actual present users need, so please manually reopen any issues you personally care about from that list in the new repo. Thanks.

Okay.

**Comment by rubyFeedback** [comment]
I think it may help if:

a) a focus on existing issues that have a patch may be prioritized

and

b) people may find libui-ng

In particular via a) that could potentially work. If anyone remembers
htop, which was created by hisham. Eventually he didn't have time
to continue with it, other folks provided some patch sets and the new
htop originated out of these patches. And I think it may be fine to
focus on the existing patches and perhaps improve a few things -
people will have a look once a project can move again. I guess
we may help cody271 reorganize and see which patches are
available and wait to be applied. Right now libui-ng is very
young - I guess tons of people don't even know it exists. I found
it only accidentally myself when searching for libui-images,
suddenly I found libui-ng. :)

**Comment by enderger** [comment]
> I think it may help if:
> 
> a) a focus on existing issues that have a patch may be prioritized
> 
> and
> 
> b) people may find libui-ng
> 
> In particular via a) that could potentially work. If anyone remembers htop, which was created by hisham. Eventually he didn't have time to continue with it, other folks provided some patch sets and the new htop originated out of these patches. And I think it may be fine to focus on the existing patches and perhaps improve a few things - people will have a look once a project can move again. I guess we may help cody271 reorganize and see which patches are available and wait to be applied. Right now libui-ng is very young - I guess tons of people don't even know it exists. I found it only accidentally myself when searching for libui-images, suddenly I found libui-ng. :)

I've created an issue (#549) to hopefully aid with part b.
